engine/include/tsengine/math.hpp

```diff
@@ -80,6 +80,185 @@ struct Matrix : BaseMatrix
     static constexpr size_t colsNum{matColumns};
 };
 
+struct Mat2 : public Matrix<2, 2>
```

Comment:
I think that now we can remove inheritance through mimic the glm by doing template specialization.
If you don't understand the comment, please let me know in dm.

Reply:
As we discussed:
I have changed my mind, we stay with it, but we should remove BaseMatrix, because this isn't longer needed